### PR TITLE
Feature: Rabbit the Fish GUI Close Blocker

### PR DESCRIPTION
**Pull request by DavidArthurCole**
## What
Adds a config option for blocking closure of the Meal Egg UI if Rabbit the Fish is present.
Also moved the Warp Menu stuff into its own accordion since there was more than one slider for the same feature-set.

## Changelog New Features
+ Added the ability to prevent closing Meal Eggs that have Rabbit the Fish inside. - Daveed

## Changelog Technical Details
+ Moved Hoppity Warp Menu config to `HoppityWarpMenuConfig`. - Daveed

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityRabbitTheFishChecker.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    /**
+     * REGEX-TEST: §cRabbit the Fish
+     */
+    private val rabbitTheFishItemPattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "item.rabbitthefish",
+        "(?:§.)*Rabbit the Fish",
+    )
+
+    /**
+     *
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityRabbitTheFishChecker.kt]: this seems to be missing

**Review by CalMWolfs — CHANGES_REQUESTED** [review]

**Review by CalMWolfs — APPROVED** [review]
code lgtm

File: src/main/java/at/hannibal2/skyhanni/mixins/transformers/gui/MixinGuiContainer.java [hunk 2]
```diff
@@ -26,6 +27,14 @@ private void closeWindowPressed(CallbackInfo ci) {
         skyHanni$hook.closeWindowPressed(ci);
     }
 
+    @Inject(method = "keyTyped", at = @At("HEAD"), cancellable = true)
+    private void onKeyTyped(char typedChar, int keyCode, CallbackInfo ci) {
+        if (!HoppityRabbitTheFishChecker.shouldContinueWithKeypress(keyCode)) {
+            ci.cancel();
+            HoppityRabbitTheFishChecker.warn();
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/mixins/transformers/gui/MixinGuiContainer.java]: move the warn function into the shouldContinueWithKeypress function

File: src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityRabbitTheFishChecker.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    fun onBackgroundDrawn(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!isEnabled()) return
+
+        rabbitTheFishIndex?.let {
+            InventoryUtils.getItemsInOpenChest()[it] highlight LorenzColor.RED
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryUpdate(event: InventoryUpdatedEvent) {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityRabbitTheFishChecker.kt]: is there a reason we use inventory update over inventory fully opdated event here?

**Review by hannibal002 — CHANGES_REQUESTED** [review]
could not test in game, but code looks good

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**